torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -958,9 +989,19 @@ void ProcessGroupNCCL::logWorkEnd(WorkNCCL& work) {
 
 void ProcessGroupNCCL::workCleanupLoop() {
   bool done = false;
-
   std::list<ProcessGroupNCCL::WorkNCCL> completedWorkList;
   while (!done || !terminateProcessGroup_.load()) {
+    struct itimerspec timerspec;
+    std::memset(&timerspec, 0, sizeof(timerspec));
+    timerspec.it_value.tv_sec = watch_timeout_secs_;
```

Comment:
Move this un-repeated part out of the while loop? Unless `timer_settime` would overwrite `timerspec`.

Reply:
i had the same question. i guess if we can confirm that timer_settime does not override timerspec, then my pref would be to make a class variable as a const timerspec that we create once and then keep using

Reply:
Make it a class variable.